File: NetFramework/XHtmlEditor/CS/C1WordsX/Tabs/ReviewTab.cs
using System;
using System.Collections.Generic;
using System.Text;
using System.Windows.Forms;
using System.Drawing;
using C1.Win.C1Ribbon;

namespace C1WordsX
{
    class ReviewTab : C1TextEditorRibbonTab
    {
        //---------------------------------------------------------------------------
        #region ** create tab

        public ReviewTab()
        {
            this.ID = this.Text = "Review";
            Groups.Add(CreateProofingGroup());
        }
        RibbonGroup CreateProofingGroup()
        {
            RibbonGroup g = new RibbonGroup();
            g.Text = g.ID = "Proofing";
            g.HasLauncherButton = true;
            g.Items.Add(CreateButton("Spelling"));
            g.Items.Add(CreateToggleButton("ShowErrors"));
            g.Items.Add(CreateButton("SetLanguage"));
            g.Items.Add(CreateButton("WordCount"));
            return g;
        }
        RibbonGroup CreateCompareGroup()
        {
            RibbonGroup g = new RibbonGroup();
            g.Text = g.ID = "Compare";
            g.Items.Add(CreateButton("Compare"));
            g.Items.Add(CreateButton("CompareVersions"));
            return g;
        }
        RibbonGroup CreateProtectGroup()
        {
            RibbonGroup g = new RibbonGroup();
            g.Text = g.ID = "Protect";
            g.Items.Add(CreateButton("Protect"));
            return g;
        }

        #endregion

        //---------------------------------------------------------------------------
        #region ** handle commands

        internal override void HandleItemEvent(RibbonEventArgs e)
        {
            RibbonItem item = (RibbonItem)e.Item;
            if (e.EventType == RibbonEventType.Click)
            {
                switch (item.ID)
                {
                    case "Spelling":
                        Ribbon.SpellCheck();
                        break;
                    case "ShowErrors":
                        Ribbon.SetActiveSpellChecking(((RibbonToggleButton)item).Pressed);
                        break;
                    case "SetLanguage":
                        Ribbon.SetSpellingLanguage();
                        break;
                    case "WordCount":
                        using (WordCount dlg = new WordCount(Editor.Text))
                        {
                            dlg.Font = Ribbon.Font;
                            dlg.ShowDialog();
                        }
                        break;
                }
            }
        }

        #endregion
    }
}

File: NetFramework/XHtmlEditor/VB/RichTextEditor/Tabs/ReviewTab.vb
Imports System.Collections.Generic
Imports System.Text
Imports System.Windows.Forms
Imports System.Drawing
Imports C1.Win.C1Ribbon

Namespace RichTextEditor
	Class ReviewTab
		Inherits C1TextEditorRibbonTab
		'---------------------------------------------------------------------------
		#Region "** create tab"

		Public Sub New()
			Me.ID = InlineAssignHelper(Me.Text, "Review")
            Groups.Add(CreateProofingGroup())
		End Sub
		Private Function CreateProofingGroup() As RibbonGroup
			Dim g As RibbonGroup = New RibbonGroup()
			g.Text = InlineAssignHelper(g.ID, "Proofing")
			g.HasLauncherButton = True
			g.Items.Add(CreateButton("Spelling"))
			g.Items.Add(CreateToggleButton("ShowErrors"))
			g.Items.Add(CreateButton("SetLanguage"))
			g.Items.Add(CreateButton("WordCount"))
			Return g
		End Function
		Private Function CreateCompareGroup() As RibbonGroup
			Dim g As RibbonGroup = New RibbonGroup()
			g.Text = InlineAssignHelper(g.ID, "Compare")
			g.Items.Add(CreateButton("Compare"))
			g.Items.Add(CreateButton("CompareVersions"))
			Return g
		End Function
		Private Function CreateProtectGroup() As RibbonGroup
			Dim g As RibbonGroup = New RibbonGroup()
			g.Text = InlineAssignHelper(g.ID, "Protect")
			g.Items.Add(CreateButton("Protect"))
			Return g
		End Function

		#End Region

		'---------------------------------------------------------------------------
		#Region "** handle commands"

		Friend Overloads Overrides Sub HandleItemEvent(e As RibbonEventArgs)
			Dim item As RibbonItem = DirectCast(e.Item, RibbonItem)
			If e.EventType = RibbonEventType.Click Then
				Select Case item.ID
					Case "Spelling"
						Ribbon.SpellCheck()
						Exit Select
					Case "ShowErrors"
						Ribbon.SetActiveSpellChecking(DirectCast(item, RibbonToggleButton).Pressed)
						Exit Select
					Case "SetLanguage"
						Ribbon.SetSpellingLanguage()
						Exit Select
					Case "WordCount"
						Using dlg As WordCount = New WordCount(Editor.Text)
							dlg.Font = Ribbon.Font
							dlg.ShowDialog()
						End Using
						Exit Select
				End Select
			End If
		End Sub
		Private Shared Function InlineAssignHelper(Of T)(ByRef target As T, value As T) As T
			target = value
			Return value
		End Function

		#End Region
	End Class
End Namespace
